sirepo/auth_db/user.py

```diff
@@ -84,13 +84,23 @@ def is_expired(self, role):
         return r.expiration < sirepo.srtime.utc_now()
 
     def uids_of_paid_users(self):
+        return self.uids_with_roles(sirepo.auth_role.PAID_USER_ROLES)
+
+    def uids_with_role(self, role):
```

Comment:
I don't think this is useful. If need be, you could check inside uids_with_roles in the for loop:

`for r in ((roles,) if isinstance(roles, str) else roles):`

I wouldn't bother. The user can always pass `[role]`.

Reply:
8e7f059f1